modules/porous_flow/src/actions/PorousFlowFullySaturated.C

```diff
@@ -0,0 +1,157 @@
+                             "single-phase fully-saturated flow problem.  No upwinding of fluid "
+                             "flow is used, so the results may differ slightly from the "
+                             "Unsaturated Action.  To run a simulation you will also need to "
+                             "provide various other Materials for each mesh "
+                             "block, depending on your simulation type, viz: permeability, "
+                             "porosity, elasticity tensor, strain calculator, stress calculator, "
+                             "matrix internal energy, thermal conductivity, diffusivity");
+  return params;
+}
+
+PorousFlowFullySaturated::PorousFlowFullySaturated(const InputParameters & params)
+  : PorousFlowSinglePhaseBase(params)
+{
+  _objects_to_add.push_back("PorousFlowFullySaturatedDarcyFlow");
+  if (_simulation_type == TRANSIENT)
+    _objects_to_add.push_back("PorousFlowMassTimeDerivative");
+  if ((_coupling_type == HydroMechanical || _coupling_type == ThermoHydroMechanical) &&
+      _simulation_type == TRANSIENT)
+    _objects_to_add.push_back("PorousFlowMassVolumetricExpansion");
+  if (_coupling_type == ThermoHydro || _coupling_type == ThermoHydroMechanical)
+    _objects_to_add.push_back("PorousFlowFullySaturatedHeatAdvection");
+}
+
+void
+PorousFlowFullySaturated::act()
+{
+  PorousFlowSinglePhaseBase::act();
+
+  // add the kernels
+  if (_current_task == "add_kernel")
+  {
+    std::string kernel_name = "PorousFlowFullySaturated_DarcyFlow";
+    std::string kernel_type = "PorousFlowFullySaturatedDarcyFlow";
+    InputParameters params = _factory.getValidParams(kernel_type);
+    params.set<UserObjectName>("PorousFlowDictator") = _dictator_name;
+    params.set<RealVectorValue>("gravity") = _gravity;
+    params.set<std::vector<SubdomainName>>("block") = {};
+
+    for (unsigned i = 0; i < _num_mass_fraction_vars; ++i)
+    {
+      kernel_name = "PorousFlowFullySaturated_DarcyFlow" + Moose::stringify(i);
+      params.set<unsigned int>("fluid_component") = i;
+      params.set<NonlinearVariableName>("variable") = _mass_fraction_vars[i];
+      _problem->addKernel(kernel_type, kernel_name, params);
+    }
+    kernel_name = "PorousFlowFullySaturated_DarcyFlow" + Moose::stringify(_num_mass_fraction_vars);
+    params.set<unsigned int>("fluid_component") = _num_mass_fraction_vars;
+    params.set<NonlinearVariableName>("variable") = _pp_var;
+    _problem->addKernel(kernel_type, kernel_name, params);
+  }
+  if (_current_task == "add_kernel" && _simulation_type == TRANSIENT)
+  {
+    std::string kernel_name = "PorousFlowFullySaturated_MassTimeDerivative";
+    std::string kernel_type = "PorousFlowMassTimeDerivative";
+    InputParameters params = _factory.getValidParams(kernel_type);
+    params.set<UserObjectName>("PorousFlowDictator") = _dictator_name;
+    params.set<std::vector<SubdomainName>>("block") = {};
```

Comment:
i was wrong: the "block = 0" problem and consequent necessity of having to set "block = {}" in Actions appears to have been fixed.  Removing all such lines in my Actions.